awd/deposit.py

```diff
@@ -94,19 +101,36 @@ def deposit(
     date = datetime.today().strftime("%m-%d-%Y %H:%M:%S")
     s3_client = s3.S3()
     sqs_client = sqs.SQS()
+    dynamodb_client = dynamodb.DynamoDB()
     dois = crossref.get_dois_from_spreadsheet(doi_file_path)
+    doi_items = dynamodb_client.retrieve_doi_items_from_database(doi_table)
     for doi in dois:
+        if dynamodb.doi_to_be_added(doi, doi_items):
+            dynamodb_client.add_doi_item_to_database(doi_table, doi)
+        elif dynamodb.doi_to_be_retried(doi, doi_items) is False:
+            continue
+        dynamodb_client.update_doi_item_status_in_database(doi_table, doi, "Processing")
+        dynamodb_client.update_doi_item_attempts_in_database(doi_table, doi)
         crossref_work_record = crossref.get_work_record_from_doi(metadata_url, doi)
         if crossref.is_valid_response(doi, crossref_work_record) is False:
+            dynamodb_client.update_doi_item_status_in_database(
+                doi_table, doi, "Failed, will retry"
```

Comment:
It'd be preferable to use a variable or, better, enum for this status (and for all the other statuses such as "Processing"), instead of this string appearing everywhere (and so many times) in the codebase. It's easy to make a mistake otherwise since omitting a single comma somewhere may cause issues (especially in when doing a comparison check). It also makes it easy to refactor code (figure out what the possible statuses are at a glance).